`mise use -g ruby` fails when `MISE_RUBY_BUILD_OPTS` are set

**Describe the bug**
`mise use -g ruby@3.3`fails with a `ruby-build` usage error when `MISE_RUBY_BUILD_OPTS` are set.

**To Reproduce**
1. Run `export MISE_RUBY_BUILD_OPTS="--without-libyaml-dir"` (or any other flag)
2. Run `mise use -g ruby@3.3 --verbose`
Program will fail with
```text
[DEBUG] ARGS: /Users/margaretgorguissian/.local/bin/mise use -g ruby@3.3 --verbose
[DEBUG] Config {
    Config Files: [],
}
[DEBUG] Toolset (180.212µs): 
[DEBUG] install_versions: ruby@3.3
[DEBUG] ToolRequestSet.build(47.271µs): ToolRequestSet: 

[DEBUG] $ /Users/margaretgorguissian/Library/Caches/mise/ruby/ruby-build/bin/ruby-build --version
[DEBUG] GET https://api.github.com/repos/rbenv/ruby-build/releases/latest
[DEBUG] starting new connection: https://api.github.com/
[DEBUG] GET https://api.github.com/repos/rbenv/ruby-build/releases/latest 200 OK
running ruby-build
[DEBUG] $ ~/Library/Caches/mise/ruby/ruby-build/bin/ruby-build --without-libyaml-dir --verbose 3.3.5 /Users/margaretgorguissian/.local/share/mise/installs/ruby/3.3.5
ruby-build: invalid flag 'without-libyaml-dir'
Usage: ruby-build [-kpv] <definition> <prefix> [-- <configure-args...>]
       ruby-build {--list|--definitions}
       ruby-build --version
  -l, --list      List latest stable releases for each Ruby
  --definitions   List all local definitions, including outdated ones
  --version       Show version of ruby-build
  -v, --verbose   Verbose mode: forward all build output to stdout/stderr
  -p, --patch     Apply a patch from stdin before building
  -k, --keep      Do not remove source tree after installation
  -4, --ipv4      Resolve names to IPv4 addresses only
  -6, --ipv6      Resolve names to IPv6 addresses only
[ERROR] ~/Library/Caches/mise/ruby/ruby-build/bin/ruby-build failed
Error: 
   0: ~/Library/Caches/mise/ruby/ruby-build/bin/ruby-build exited with non-zero status: exit code 1
```

**Expected behavior**
I expect the ruby v3.3.5 installation to succeed, with the provided options respected.

**`mise doctor` output**

```text
version: 2024.9.5 macos-x64 (1f0f03e 2024-09-17)
activated: yes
shims_on_path: no

build_info: 
  Target: x86_64-apple-darwin
  Features: DEFAULT, NATIVE_TLS, OPENSSL
  Built: Tue, 17 Sep 2024 13:42:17 +0000
  Rust Version: rustc 1.81.0 (eeb90cda1 2024-09-04)
  Profile: release

shell: 
  /bin/zsh
  zsh 5.9 (x86_64-apple-darwin22.0)

dirs: 
  data: ~/.local/share/mise
  config: ~/.config/mise
  cache: ~/Library/Caches/mise
  state: ~/.local/state/mise
  shims: ~/.local/share/mise/shims

config_files: 


backends: 
  cargo
  core
  go
  npm
  pipx
  spm
  ubi
  vfox

plugins: 


toolset: 


env_vars: 
  MISE_SHELL=zsh
  MISE_RUBY_BUILD_OPTS=--without-libyaml-dir

settings: 
  activate_aggressive = false
  all_compile = false
  always_keep_download = false
  always_keep_install = false
  asdf = true
  asdf_compat = false
  cargo_binstall = true
  color = true
  disable_default_shorthands = false
  disable_hints = []
  disable_tools = []
  experimental = false
  go_default_packages_file = "~/.default-go-packages"
  go_download_mirror = "https://dl.google.com/go"
  go_repo = "https://github.com/golang/go"
  go_set_gopath = false
  go_set_goroot = true
  go_skip_checksum = false
  http_timeout = 30
  jobs = 4
  legacy_version_file = true
  legacy_version_file_disable_tools = []
  libgit2 = true
  node_compile = false
  not_found_auto_install = true
  paranoid = false
  pipx_uvx = false
  plugin_autoupdate_last_check_duration = "7d"
  python_default_packages_file = "~/.default-python-packages"
  python_pyenv_repo = "https://github.com/pyenv/pyenv.git"
  raw = false
  trusted_config_paths = []
  quiet = false
  use_versions_host = true
  verbose = false
  vfox = false
  yes = false
  ci = false
  debug = false
  trace = false
  log_level = "info"
  python_venv_auto_create = false

  [status]
  missing_tools = "if_other_versions_installed"
  show_env = false
  show_tools = false

No warnings found
No problems found
```

**Additional context**
My naive suspicion is that the arguments are not being added to the `args` array correctly here:
https://github.com/jdx/mise/blob/a92e7bc9e64e65d00c14ea0a79721a9e99a80fbe/src/plugins/core/ruby.rs#L277
As articulated in the usage message when running `ruby-build`, the definition and the prefix should come _before_ any configuration arguments (`ruby-build [-kpv] <definition> <prefix> [-- <configure-args...>]`). However, the definition (version) and prefix are pushed onto the end of the args array, rather than being prepended. I can open a PR to fix this, but wanted to file an issue first (as recommended [here](https://mise.jdx.dev/contributing.html#contributing)) in case I am missing some context here. Thank you!

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -24,6 +24,7 @@ env:
   SCCACHE_S3_NO_CREDENTIALS: ${{ secrets.MINIO_AWS_ACCESS_KEY_ID && '0' || '1' }}
   MINIO_AWS_ACCESS_KEY_ID: ${{ secrets.MINIO_AWS_ACCESS_KEY_ID }}
   MINIO_AWS_SECRET_ACCESS_KEY: ${{ secrets.MINIO_AWS_SECRET_ACCESS_KEY }}
+  GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
 
 permissions:
   pull-requests: write

--- a/docs/lang/ruby.md
+++ b/docs/lang/ruby.md
@@ -27,14 +27,71 @@ for additional settings and some troubleshooting.
 
 `ruby-build` already has a
 [handful of settings](https://github.com/rbenv/ruby-build?tab=readme-ov-file#custom-build-configuration),
-in additional to that mise has a few extra configuration variables:
+in additional to that mise has a few extra settings:
 
-- `MISE_RUBY_INSTALL` [bool]: Build with ruby-install instead of ruby-build
-- `MISE_RUBY_APPLY_PATCHES` [string]: A list of patches (files or URLs) to apply to the ruby source code
-- `MISE_RUBY_VERBOSE_INSTALL` [bool]: Show verbose output during installation (passes --verbose to ruby-build)
-- `MISE_RUBY_BUILD_OPTS` [string]: Command line options to pass to ruby-build when installing
-- `MISE_RUBY_INSTALL_OPTS` [string]: Command line options to pass to ruby-install when installing (if MISE_RUBY_INSTALL=1)
-- `MISE_RUBY_DEFAULT_PACKAGES_FILE` [string]: location of default gems file, defaults to `$HOME/.default-gems`
+### `ruby.apply_patches`
+
+* Type: `Option<String>`
+* Env: `MISE_RUBY_APPLY_PATCHES`
+* Default: `None`
+
+A list of patches (files or URLs) to apply to the ruby source code.
+
+### `ruby.verbose_install`
+
+* Type: `Option<bool>`
+* Env: `MISE_RUBY_VERBOSE_INSTALL`
+* Default: `None` (behaves like `false` unless --verbose is passed to mise)
+
+Show verbose output during installation (passes --verbose to ruby-build)
+
+### `ruby.ruby_build_repo`
+
+* Type: `String`
+* Env: `MISE_RUBY_BUILD_REPO`
+* Default: `https://github.com/rbenv/ruby-build.git`
+
+The git repository to use for getting ruby-build.
+
+### `ruby.ruby_build_opts`
+
+* Type: `Option<String>`
+* Env: `MISE_RUBY_BUILD_OPTS`
+* Default: `None`
+
+Command line options to pass to ruby-build when installing
+
+### `ruby.ruby_install`
+
+* Type: `bool`
+* Env: `MISE_RUBY_INSTALL`
+* Default: `false`
+
+Build with [ruby-install](https://github.com/postmodern/ruby-install) instead of [ruby-build](https://github.com/rbenv/ruby-build).
+
+### `ruby.ruby_install_repo`
+
+* Type: `String`
+* Env: `MISE_RUBY_INSTALL_REPO`
+* Default: `https://github.com/postmodern/ruby-install.git`
+
+The git repository to use for getting ruby-install.
+
+### `ruby.ruby_install_opts`
+
+* Type: `Option<String>`
+* Env: `MISE_RUBY_INSTALL_OPTS`
+* Default: `None`
+
+Command line options to pass to ruby-install when installing (if MISE_RUBY_INSTALL=1)
+
+### `ruby.default_packages_file`
+
+* Type: `String`
+* Env: `MISE_RUBY_DEFAULT_PACKAGES_FILE`
+* Default: `$HOME/.default-gems`
+
+Location of file containing a list of gems to install after installing a new ruby version.
 
 ## Default gems
 

--- a/schema/mise.json
+++ b/schema/mise.json
@@ -429,6 +429,45 @@
           "description": "path to file containing shorthand mappings",
           "type": "string"
         },
+        "ruby": {
+          "description": "settings specific to ruby",
+          "type": "object",
+          "additionalProperties": false,
+          "properties": {
+            "apply_patches": {
+              "description": "A list of patches (files or URLs) to apply to Ruby source code before building",
+              "type": "string"
+            },
+            "verbose_install": {
+              "description": "Show verbose output during installation (passes --verbose to ruby-build)",
+              "type": "boolean"
+            },
+            "ruby_build_repo": {
+              "description": "The git repository to use for getting ruby-build",
+              "type": "string"
+            },
+            "ruby_build_opts": {
+              "description": "Command line options to pass to ruby-build when installing",
+              "type": "string"
+            },
+            "ruby_install": {
+              "description": "Build with ruby-install instead of ruby-build",
+              "type": "boolean"
+            },
+            "ruby_install_repo": {
+              "description": "The git repository to use for getting ruby-install",
+              "type": "string"
+            },
+            "ruby_install_opts": {
+              "description": "Command line options to pass to ruby-install when installing (if MISE_RUBY_INSTALL=1)",
+              "type": "string"
+            },
+            "default_packages_file": {
+              "description": "Location of file containing a list of gems to install after installing a new ruby version.",
+              "type": "string"
+            }
+          }
+        },
         "status": {
           "description": "configure messages displayed when changing directories or executing tools",
           "type": "object",

--- a/src/cli/settings/ls.rs
+++ b/src/cli/settings/ls.rs
@@ -58,7 +58,7 @@ mod tests {
     #[test]
     fn test_settings_ls() {
         reset();
-        assert_cli_snapshot!("settings", @r###"
+        assert_cli_snapshot!("settings", @r#"
         activate_aggressive = false
         all_compile = false
         always_keep_download = true
@@ -97,17 +97,23 @@ mod tests {
         vfox = false
         yes = true
 
+        [ruby]
+        default_packages_file = "~/.default-gems"
+        ruby_build_repo = "https://github.com/rbenv/ruby-build.git"
+        ruby_install = false
+        ruby_install_repo = "https://github.com/postmodern/ruby-install.git"
+
         [status]
         missing_tools = "if_other_versions_installed"
         show_env = false
         show_tools = false
-        "###);
+        "#);
     }
 
     #[test]
     fn test_settings_ls_keys() {
         reset();
-        assert_cli_snapshot!("settings", "--keys", @r###"
+        assert_cli_snapshot!("settings", "--keys", @r#"
         activate_aggressive
         all_compile
         always_keep_download
@@ -140,6 +146,11 @@ mod tests {
         python_pyenv_repo
         quiet
         raw
+        ruby
+        ruby.default_packages_file
+        ruby.ruby_build_repo
+        ruby.ruby_install
+        ruby.ruby_install_repo
         status
         status.missing_tools
         status.show_env
@@ -149,6 +160,6 @@ mod tests {
         verbose
         vfox
         yes
-        "###);
+        "#);
     }
 }

--- a/src/cli/settings/set.rs
+++ b/src/cli/settings/set.rs
@@ -134,7 +134,7 @@ pub mod tests {
             "1"
         );
 
-        assert_cli_snapshot!("settings", @r###"
+        assert_cli_snapshot!("settings", @r#"
         activate_aggressive = false
         all_compile = false
         always_keep_download = true
@@ -173,11 +173,17 @@ pub mod tests {
         vfox = false
         yes = true
 
+        [ruby]
+        default_packages_file = "~/.default-gems"
+        ruby_build_repo = "https://github.com/rbenv/ruby-build.git"
+        ruby_install = false
+        ruby_install_repo = "https://github.com/postmodern/ruby-install.git"
+
         [status]
         missing_tools = "never"
         show_env = false
         show_tools = false
-        "###);
+        "#);
         reset();
     }
 }

--- a/src/cli/settings/unset.rs
+++ b/src/cli/settings/unset.rs
@@ -49,7 +49,7 @@ mod tests {
 
         assert_cli!("settings", "unset", "jobs");
 
-        assert_cli_snapshot!("settings", @r###"
+        assert_cli_snapshot!("settings", @r#"
         activate_aggressive = false
         all_compile = false
         always_keep_download = true
@@ -88,11 +88,17 @@ mod tests {
         vfox = false
         yes = true
 
+        [ruby]
+        default_packages_file = "~/.default-gems"
+        ruby_build_repo = "https://github.com/rbenv/ruby-build.git"
+        ruby_install = false
+        ruby_install_repo = "https://github.com/postmodern/ruby-install.git"
+
         [status]
         missing_tools = "if_other_versions_installed"
         show_env = false
         show_tools = false
-        "###);
+        "#);
 
         reset();
     }

--- a/src/config/settings.rs
+++ b/src/config/settings.rs
@@ -4,6 +4,9 @@ use std::iter::once;
 use std::path::PathBuf;
 use std::sync::{Arc, Mutex, RwLock};
 
+use crate::config::{system_config_files, DEFAULT_CONFIG_FILENAMES};
+use crate::file::FindUp;
+use crate::{config, dirs, env, file};
 #[allow(unused_imports)]
 use confique::env::parse::{list_by_colon, list_by_comma};
 use confique::{Config, Partial};
@@ -11,10 +14,6 @@ use eyre::{bail, Result};
 use once_cell::sync::Lazy;
 use serde::ser::Error;
 use serde_derive::{Deserialize, Serialize};
-
-use crate::config::{system_config_files, DEFAULT_CONFIG_FILENAMES};
-use crate::file::FindUp;
-use crate::{config, dirs, env, file};
 
 #[rustfmt::skip]
 #[derive(Config, Default, Debug, Clone, Serialize)]
@@ -118,6 +117,8 @@ pub struct Settings {
     pub python_pyenv_repo: String,
     #[config(env = "MISE_RAW", default = false)]
     pub raw: bool,
+    #[config(nested)]
+    pub ruby: SettingsRuby,
     #[config(env = "MISE_SHORTHANDS_FILE")]
     pub shorthands_file: Option<PathBuf>,
     /// what level of status messages to display when entering directories
@@ -157,6 +158,29 @@ pub struct Settings {
     pub log_level: String,
     #[config(env = "MISE_PYTHON_VENV_AUTO_CREATE", default = false)]
     pub python_venv_auto_create: bool,
+}
+
+#[derive(Config, Default, Debug, Clone, Serialize)]
+#[config(partial_attr(derive(Clone, Serialize, Default)))]
+#[config(partial_attr(serde(deny_unknown_fields)))]
+#[rustfmt::skip]
+pub struct SettingsRuby {
+    #[config(env = "MISE_RUBY_APPLY_PATCHES")]
+    pub apply_patches: Option<String>,
+    #[config(env = "MISE_RUBY_DEFAULT_PACKAGES_FILE", default = "~/.default-gems")]
+    pub default_packages_file: String,
+    #[config(env = "MISE_RUBY_BUILD_REPO", default = "https://github.com/rbenv/ruby-build.git")]
+    pub ruby_build_repo: String,
+    #[config(env = "MISE_RUBY_BUILD_OPTS")]
+    pub ruby_build_opts: Option<String>,
+    #[config(env = "MISE_RUBY_INSTALL", default = false)]
+    pub ruby_install: bool,
+    #[config(env = "MISE_RUBY_INSTALL_REPO", default = "https://github.com/postmodern/ruby-install.git")]
+    pub ruby_install_repo: String,
+    #[config(env = "MISE_RUBY_INSTALL_OPTS")]
+    pub ruby_install_opts: Option<String>,
+    #[config(env = "MISE_RUBY_VERBOSE_INSTALL")]
+    pub verbose_install: Option<bool>,
 }
 
 #[derive(Config, Default, Debug, Clone, Serialize)]

--- a/src/env.rs
+++ b/src/env.rs
@@ -232,34 +232,6 @@ pub static NVM_DIR: Lazy<PathBuf> =
 pub static NODENV_ROOT: Lazy<PathBuf> =
     Lazy::new(|| var_path("NODENV_ROOT").unwrap_or_else(|| HOME.join(".nodenv")));
 
-// ruby
-#[cfg(unix)]
-pub static MISE_RUBY_BUILD_REPO: Lazy<String> = Lazy::new(|| {
-    var("MISE_RUBY_BUILD_REPO").unwrap_or_else(|_| "https://github.com/rbenv/ruby-build.git".into())
-});
-#[cfg(unix)]
-pub static MISE_RUBY_INSTALL_REPO: Lazy<String> = Lazy::new(|| {
-    var("MISE_RUBY_INSTALL_REPO")
-        .unwrap_or_else(|_| "https://github.com/postmodern/ruby-install.git".into())
-});
-#[cfg(unix)]
-pub static MISE_RUBY_INSTALL: Lazy<bool> = Lazy::new(|| var_is_true("MISE_RUBY_INSTALL"));
-#[cfg(unix)]
-pub static MISE_RUBY_APPLY_PATCHES: Lazy<Option<String>> =
-    Lazy::new(|| var("MISE_RUBY_APPLY_PATCHES").ok());
-#[cfg(unix)]
-pub static MISE_RUBY_VERBOSE_INSTALL: Lazy<Option<bool>> =
-    Lazy::new(|| var_option_bool("MISE_RUBY_VERBOSE_INSTALL"));
-#[cfg(unix)]
-pub static MISE_RUBY_INSTALL_OPTS: Lazy<Result<Vec<String>, shell_words::ParseError>> =
-    Lazy::new(|| shell_words::split(&var("MISE_RUBY_INSTALL_OPTS").unwrap_or_default()));
-#[cfg(unix)]
-pub static MISE_RUBY_BUILD_OPTS: Lazy<Result<Vec<String>, shell_words::ParseError>> =
-    Lazy::new(|| shell_words::split(&var("MISE_RUBY_BUILD_OPTS").unwrap_or_default()));
-pub static MISE_RUBY_DEFAULT_PACKAGES_FILE: Lazy<PathBuf> = Lazy::new(|| {
-    var_path("MISE_RUBY_DEFAULT_PACKAGES_FILE").unwrap_or_else(|| HOME.join(".default-gems"))
-});
-
 fn get_env_diff() -> EnvDiff {
     let env = vars().collect::<HashMap<_, _>>();
     match env.get("__MISE_DIFF") {

--- a/src/plugins/core/ruby.rs
+++ b/src/plugins/core/ruby.rs
@@ -49,7 +49,8 @@ impl RubyPlugin {
     }
 
     fn lock_build_tool(&self) -> Result<fslock::LockFile> {
-        let build_tool_path = if *env::MISE_RUBY_INSTALL {
+        let settings = Settings::get();
+        let build_tool_path = if settings.ruby.ruby_install {
             self.ruby_build_bin()
         } else {
             self.ruby_install_bin()
@@ -62,7 +63,8 @@ impl RubyPlugin {
     }
 
     fn update_build_tool(&self) -> Result<()> {
-        if *env::MISE_RUBY_INSTALL {
+        let settings = Settings::get();
+        if settings.ruby.ruby_install {
             self.update_ruby_install()
                 .wrap_err("failed to update ruby-install")?;
         }
@@ -71,6 +73,7 @@ impl RubyPlugin {
     }
 
     fn install_ruby_build(&self) -> Result<()> {
+        let settings = Settings::get();
         debug!(
             "Installing ruby-build to {}",
             self.ruby_build_path().display()
@@ -79,7 +82,7 @@ impl RubyPlugin {
         file::remove_all(&tmp)?;
         file::create_dir_all(tmp.parent().unwrap())?;
         let git = Git::new(tmp.clone());
-        git.clone(&env::MISE_RUBY_BUILD_REPO)?;
+        git.clone(&settings.ruby.ruby_build_repo)?;
 
         cmd!("sh", "install.sh")
             .env("PREFIX", self.ruby_build_path())
@@ -112,6 +115,7 @@ impl RubyPlugin {
     }
 
     fn install_ruby_install(&self) -> Result<()> {
+        let settings = Settings::get();
         debug!(
             "Installing ruby-install to {}",
             self.ruby_install_path().display()
@@ -120,7 +124,7 @@ impl RubyPlugin {
         file::remove_all(&tmp)?;
         file::create_dir_all(tmp.parent().unwrap())?;
         let git = Git::new(tmp.clone());
-        git.clone(&env::MISE_RUBY_INSTALL_REPO)?;
+        git.clone(&settings.ruby.ruby_install_repo)?;
 
         cmd!("make", "install")
             .env("PREFIX", self.ruby_install_path())
@@ -184,7 +188,9 @@ impl RubyPlugin {
         tv: &ToolVersion,
         pr: &dyn SingleReport,
     ) -> Result<()> {
-        let body = file::read_to_string(&*env::MISE_RUBY_DEFAULT_PACKAGES_FILE).unwrap_or_default();
+        let settings = Settings::get();
+        let default_gems_file = file::replace_path(&settings.ruby.default_packages_file);
+        let body = file::read_to_string(&default_gems_file).unwrap_or_default();
         for package in body.lines() {
             let package = package.split('#').next().unwrap_or_default().trim();
             if package.is_empty() {
@@ -257,7 +263,8 @@ impl RubyPlugin {
         tv: &ToolVersion,
         pr: &'a dyn SingleReport,
     ) -> Result<CmdLineRunner<'a>> {
-        let cmd = if *env::MISE_RUBY_INSTALL {
+        let settings = Settings::get();
+        let cmd = if settings.ruby.ruby_install {
             CmdLineRunner::new(self.ruby_install_bin()).args(self.install_args_ruby_install(tv)?)
         } else {
             CmdLineRunner::new(self.ruby_build_bin())
@@ -267,18 +274,24 @@ impl RubyPlugin {
         Ok(cmd.with_pr(pr).envs(config.env()?))
     }
     fn install_args_ruby_build(&self, tv: &ToolVersion) -> Result<Vec<String>> {
-        let mut args = env::MISE_RUBY_BUILD_OPTS.clone()?;
+        let settings = Settings::get();
+        let mut args = vec![];
         if self.verbose_install() {
             args.push("--verbose".into());
         }
-        if env::MISE_RUBY_APPLY_PATCHES.is_some() {
+        if settings.ruby.apply_patches.is_some() {
             args.push("--patch".into());
         }
         args.push(tv.version.clone());
         args.push(tv.install_path().to_string_lossy().to_string());
+        if let Some(opts) = &settings.ruby.ruby_build_opts {
+            args.push("--".into());
+            args.extend(shell_words::split(opts)?);
+        }
         Ok(args)
     }
     fn install_args_ruby_install(&self, tv: &ToolVersion) -> Result<Vec<String>> {
+        let settings = Settings::get();
         let mut args = vec![];
         for patch in self.fetch_patch_sources() {
             args.push("--patch".into());
@@ -292,18 +305,22 @@ impl RubyPlugin {
         args.push(version.into());
         args.push("--install-dir".into());
         args.push(tv.install_path().to_string_lossy().to_string());
-        args.extend(env::MISE_RUBY_INSTALL_OPTS.clone()?);
+        if let Some(opts) = &settings.ruby.ruby_install_opts {
+            args.push("--".into());
+            args.extend(shell_words::split(opts)?);
+        }
         Ok(args)
     }
 
     fn verbose_install(&self) -> bool {
         let settings = Settings::get();
-        let verbose_env = *env::MISE_RUBY_VERBOSE_INSTALL;
+        let verbose_env = settings.ruby.verbose_install;
         verbose_env == Some(true) || (settings.verbose && verbose_env != Some(false))
     }
 
     fn fetch_patch_sources(&self) -> Vec<String> {
-        let patch_sources = env::MISE_RUBY_APPLY_PATCHES.clone().unwrap_or_default();
+        let settings = Settings::get();
+        let patch_sources = settings.ruby.apply_patches.clone().unwrap_or_default();
         patch_sources
             .split('\n')
             .map(|s| s.to_string())

--- a/src/plugins/core/ruby_windows.rs
+++ b/src/plugins/core/ruby_windows.rs
@@ -4,7 +4,7 @@ use std::path::{Path, PathBuf};
 use crate::backend::Backend;
 use crate::cli::args::BackendArg;
 use crate::cmd::CmdLineRunner;
-use crate::config::Config;
+use crate::config::{Config, Settings};
 use crate::env::PATH_KEY;
 use crate::github::GithubRelease;
 use crate::http::HTTP;
@@ -67,7 +67,9 @@ impl RubyPlugin {
         tv: &ToolVersion,
         pr: &dyn SingleReport,
     ) -> Result<()> {
-        let body = file::read_to_string(&*env::MISE_RUBY_DEFAULT_PACKAGES_FILE).unwrap_or_default();
+        let settings = Settings::get();
+        let default_gems_file = file::replace_path(&settings.ruby.default_packages_file);
+        let body = file::read_to_string(&default_gems_file).unwrap_or_default();
         for package in body.lines() {
             let package = package.split('#').next().unwrap_or_default().trim();
             if package.is_empty() {

